frontend/src/pages/Home.js

```diff
@@ -1,28 +1,35 @@
-import React, { useState, useEffect } from "react"
+import React, { useEffect } from "react"
+import { useDispatch, useSelector } from "react-redux"
 import { Link } from "react-router-dom"
+import Loader from "../components/partials/loader/Loader"
 import ProductCard from "../components/partials/ProductCard/ProductCard"
-import { api } from "../utils/Api"
+import { productsList } from "../store/products/actions"
 
 export default function Home() {
-    const [products, setProducts] = useState([])
+    const dispatch = useDispatch()
+    const { products, error, isLoading } = useSelector(state => state.productList)
 
     useEffect(() => {
-        api("api/products", {}, "get").then(({ data }) => {
-            setProducts(data)
-        })
-    }, [])
+        dispatch(productsList())
+
+    }, [dispatch])
     return (
-        <div className="container d-flex flex-wrap" > {
-            products.map(product => (
-                <div key={product._id}
-                    className="col-12 col-md-6 col-lg-3 m-y-15" >
-                    <Link className="no-decoration txt-color-main-500"
-                        to={`/products/${product._id}`} >
-                        <ProductCard product={product}
-                        />
-                    </Link>
-                </div>
-            ))
-        } </div>
+        <div className="w-100">
+            {isLoading ? <Loader /> :
+
+                (error ? <h3 className="txt-color-danger-500 fs-16">{error}</h3> : <div className="container d-flex flex-wrap" > {
```

Comment:
avoid using nested ternary operator (use if statements to assign the element should be rendered to a variable before the return() and use the variable in the JSX)